spec/requests/api/v2025_0/put_trainee_spec.rb

```diff
@@ -478,11 +479,7 @@
             let(:new_lead_partner) { build(:lead_partner, :school) }
 
             it "sets lead_partner_urn to nil and lead_partner_not_applicable to true" do
-              expect(response.parsed_body[:data][:lead_partner_urn]).to be_nil
-
-              trainee.reload
-
-              expect(trainee.lead_partner_not_applicable).to be(true)
+              expect(response).to have_http_status(:unprocessable_entity)
```

Comment:
I think we should test that the trainee values in the db haven't been modified, as well as the exact error message in the response.